chat-components/src/services/EventQueue.ts

```diff
@@ -12,20 +12,20 @@ class EventQueue {
         this.newMessage = newMessage;
     }
 
-    processEvents(deferTimeout = 0) {
+    processEvents() {
         const dequeue = () => {
             this.channelEventQueue.forEach((event, eventId) => { // Process entry based on insertion order
                 this.newMessage.next(event); // Post event directly instead of using pubChannel
                 this.channelEventQueue.delete(eventId); // Remove event from queue regardless of outcome
             });
         };
 
-        setTimeout(() => {
-            dequeue();
-        }, deferTimeout);
+        dequeue();
     }
 
     queueEvents(timeout = 500) {
+        this.stopIfEmpty();

```

Comment:
isnt this duplicate, stopIfEmpty is already doing this validation

Reply:
Logic removed